### ISSUE TITLE
Hosts stop running or archiving tests

### ISSUE THREAD

**Issue by rhclopes** [perfsonar/psconfig#112]
Hi,

We have seen an increasing number of hosts that stop running tests or at least arching and publishing results. See for example,

https://psmad.opensciencegrid.org/maddash-webui/index.cgi?dashboard=UK%20Mesh%20Config

Some sysadmins have resorted to restarting services or rebooting hosts. However, frequently, even that will not return the host to normal behaviour. Some admins have reported reboot the host two or three times before they sees running and being reported. I have tested a  case where five reboots were not enough to restore services to normality.

For example, I have seen the following timeline with  the host 'Jisc London' in the link above:

(0) host had completely stopped showing any throughput results.
(1) troubleshooting showed errors only idle test, see [0]
(2) rebooting the host several times didn't produce any improvement: 'pscheduler troubleshoot' still failed as in [0]
(3) stop services with script like [1] and start again didn't show any improvement
(4) stop services (wiith [1]) and list open ports (with netstat) and list process with `ps aux` that logstash, memcache, twampd and toolkit_config were still alive.
(5) kill all those (logstash, memcache, twampd, toolkit_config) and start services with [2] got all measurements back and troubleshoot working [3]
 

Regards,
Raul

[0] pcheduler troubleshot

[raul@ps-slough-10g ~]$ echo $lon
ps-london-bw.ja.net
[raul@ps-slough-10g ~]$ echo $slough
ps-slough-10g.ja.net

[raul@ps-slough-10g ~]$  pscheduler troubleshoot --host  $lon  $slough
Performing basic troubleshooting of ps-london-bw.ja.net and ps-slough-10g.ja.net.

ps-london-bw.ja.net:

  Measuring MTU... 9000+
  Looking for pScheduler... OK.
  Fetching API level... 5
  Checking clock... OK.
  Exercising API... Archivers... Contexts... Tests... Tools... OK.
  Fetching service status... OK.
  Checking services... Ticker... Scheduler... Runner... Archiver... OK.
  Checking limits... OK.
  Idle test.... 9 seconds... Missed... Failed.

Did not get a result: Resource Not found.
```
[1] stop-services

#!/bin/bash

systemctl stop httpd
	
systemctl stop pscheduler-archiver.service
	
systemctl stop pscheduler-runner.service

systemctl stop pscheduler-scheduler.service

systemctl stop pscheduler-ticker.service

systemctl stop psconfig-pscheduler-agent.service

systemctl stop psconfig-pscheduler-agent.service

systemctl stop perfsonar-lscachedaemon.service

systemctl stop postgresql-10.service

systemctl stop opensearch

systemctl stop logstash

systemctl stop elmond

[2] start services

systemctl restart httpd
systemctl restart pscheduler-archiver.service
systemctl restart pscheduler-runner.service
systemctl restart pscheduler-scheduler.service
systemctl restart pscheduler-ticker.service
systemctl restart psconfig-pscheduler-agent.service
systemctl restart psconfig-pscheduler-agent.service
systemctl restart psconfig-pscheduler-agent.service
systemctl restart perfsonar-lscachedaemon.service
systemctl restart postgresql-10.service
systemctl restart opensearch
systemctl restart logstash
systemctl restart elmond

[3] troubleshoot after killing all services (by 'kill -9 pid') and restarting with [2]

Performing basic troubleshooting of ps-london-bw.ja.net and ps-slough-10g.ja.net.

ps-london-bw.ja.net:

  Measuring MTU... 9000+
  Looking for pScheduler... OK.
  Fetching API level... 5
  Checking clock... OK.
  Exercising API... Archivers... Contexts... Tests... Tools... OK.
  Fetching service status... OK.
  Checking services... Ticker... Scheduler... Runner... Archiver... OK.
  Checking limits... OK.
  Idle test.... 9 seconds... Finished... Checking archiving... OK.

ps-slough-10g.ja.net:

  Measuring MTU... 65535 (Local)
  Looking for pScheduler... OK.
  Fetching API level... 5
  Checking clock... OK.
  Exercising API... Archivers... Contexts... Tests... Tools... OK.
  Fetching service status... OK.
  Checking services... Ticker... Scheduler... Runner... Archiver... OK.
  Checking limits... OK.
  Idle test.... 9 seconds... Finished... Checking archiving... OK.

ps-london-bw.ja.net and ps-slough-10g.ja.net:

  Checking IP addresses... OK.
  Measuring MTU... 9000+
  Checking timekeeping... OK.
  Simple stream test.... 12 seconds... Finished... OK.

pScheduler on both hosts appears to be functioning normally.

**Comment by tonin** [comment]
Some information: this happened on EL7 hosts and continues to happen now with 5.0.5.

To be able to support you further, @rhclopes it would be helpful to have more information when this happens, like:
- put pscheduler in debug mode
- run a manual CLI pscheduler command and get the output
- turn debug mode off to avoid filling up the logs
- send us the output of the log file for this specific CLI run with corresponding timestamps

**Comment by rhclopes** [comment]
Please find attached output of `pscheduler diags` for a host that stop running perfSONAR services. Timeline:

(0) Host was running until a couple of weeks.
(1) Today user asked for my help.
(2) No services running, troublesoot failed with
```
                                                                                                         [351/1980]
[psremote@ps08-em1 ~]$ pscheduler troubleshoot --host ps08-em1.rl.ac.uk ps07-em1.rl.ac.uk                                                                                   
Performing basic troubleshooting of ps08-em1.rl.ac.uk and ps07-em1.rl.ac.uk.

ps08-em1.rl.ac.uk:

  Measuring MTU... 65535 (Local)
  Looking for pScheduler... OK.
  Fetching API level... 5
  Checking clock... OK.
  Exercising API... Archivers... Contexts... Tests... Tools... OK.
  Fetching service status... OK.
  Checking services... Ticker... Scheduler... Runner... Archiver... OK.
  Checking limits... OK.
  Idle test.... 4 seconds... Non-Starter... Failed.

Did not get a result: Resource Not found.
(3) I enabled `sudo pscheduler debug on` and
 - tried throughput test: failed with
   ```
[psremote@ps08-em1 ~]$ pscheduler task throughput --source ps08-ban.rl.ac.uk --dest ps07-ban.rl.ac.uk                                                                       
Submitting task...
Task URL:
https://ps08-ban.rl.ac.uk/pscheduler/tasks/839a297a-f8b8-4626-b0bb-4ec043e188af
Running with tool 'iperf3'
Fetching first run...

Next scheduled run:
https://ps08-ban.rl.ac.uk/pscheduler/tasks/839a297a-f8b8-4626-b0bb-4ec043e188af/runs/26c67d28-389a-486a-94fb-8edfed06e84f

Run scheduled at 2023-10-25T09:23:12+00:00 is a non-starter:
Error fetching task: 
```

(4) I ran `sudo pscheduler diags > diags.out`. Please find attached
(5) stopped the service with `sudo stop.sh` 
[start-stop.tar.gz](https://github.com/perfsonar/psconfig/files/13164491/start-stop.tar.gz)
[start-stop.tar.gz](https://github.com/perfsonar/psconfig/files/13164502/start-stop.tar.gz)

(6) started the services with `sudo sh start.sh` 
(7) I ran `sudo pscheduler diags > diags-after-restart.out` (attached)
(8) `sudo pscheduler debug off`

Host is back running.
[diags.txt](https://github.com/perfsonar/psconfig/files/13164424/diags.txt)
[diags-after-restart.txt](https://github.com/perfsonar/psconfig/files/13164428/diags-after-restart.txt)

```

**Comment by arlake228** [comment]
Closing since we included what appears to be a fix in 5.0.6. Will revisit if behavior resurfaces after releases.
